src/main/java/io/cdap/plugin/db/batch/action/vertica/load/VerticaImportConfig.java

```diff
@@ -0,0 +1,300 @@
+    this.delimiter = delimiter;
+    this.copyStatement = copyStatement;
+    this.path = path;
+    this.autoCommit = autoCommit;
+  }
+
+  private VerticaImportConfig(Builder builder) {
+    connectionString = builder.connectionString;
+    user = builder.user;
+    password = builder.password;
+    level = builder.level;
+    tableName = builder.tableName;
+    delimiter = builder.delimiter;
+    copyStatement = builder.copyStatement;
+    path = builder.path;
+    autoCommit = builder.autoCommit;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(VerticaImportConfig copy) {
+    return builder()
+      .setConnectionString(copy.connectionString)
+      .setUser(copy.user)
+      .setPassword(copy.password)
+      .setLevel(copy.level)
+      .setTableName(copy.tableName)
+      .setDelimiter(copy.delimiter)
+      .setCopyStatement(copy.copyStatement)
+      .setPath(copy.path)
+      .setAutoCommit(copy.autoCommit);
+  }
+
+  public String getConnectionString() {
+    return connectionString;
+  }
+
+  @Nullable
+  public String getUser() {
+    return user;
+  }
+
+  @Nullable
+  public String getPassword() {
+    return password;
+  }
+
+  public String getLevel() {
+    return level;
+  }
+
+  @Nullable
+  public String getTableName() {
+    return tableName;
+  }
+
+  @Nullable
+  public String getDelimiter() {
+    return delimiter;
+  }
+
+  @Nullable
+  public String getCopyStatement() {
+    return copyStatement;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  public String getAutoCommit() {
+    return autoCommit;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    if(!containsMacro(CONNECTION_STRING)) {
+      if (Strings.isNullOrEmpty(connectionString)) {
+        failureCollector.addFailure(
+          "Connection String must be provided.", null)
+          .withConfigProperty(CONNECTION_STRING);
+      } else if (!connectionString.startsWith(CONNECTION_STRING_PREFIX)) {
+        failureCollector.addFailure(
+          "Invalid connection string.",
+          "Connection String must comply with format - jdbc:vertica://<VerticaHost>:<portNumber>/<databaseName>")
+          .withConfigProperty(CONNECTION_STRING);
+      }
+    }
+    if (!containsMacro(USER) && !containsMacro(PASSWORD)) {
+      if (Strings.isNullOrEmpty(user) && !Strings.isNullOrEmpty(password)) {
+        failureCollector.addFailure(
+          "Username is not specified.",
+          "Ensure both username and password are provided.")
+          .withConfigProperty(USER).withConfigProperty(PASSWORD);
+      }
+      if (!Strings.isNullOrEmpty(user) && Strings.isNullOrEmpty(password)) {
+        failureCollector.addFailure(
+          "Password is not specified.",
+          "Ensure both username and password are provided.")
+          .withConfigProperty(USER).withConfigProperty(PASSWORD);
+      }
+    }
+    if (!containsMacro(USER)) {
```

Comment:
why are we checking for user is a macro here?

Reply:
was corrected to !containsMacro(LEVEL)